src/nvim/highlight_group.c

```diff
@@ -824,7 +825,7 @@ void set_hl_group(int id, HlAttrs attrs, Dict(highlight) *dict, int link_id)
     g->sg_link = 0;
   }
 
-  g->sg_gui = attrs.rgb_ae_attr;
+  g->sg_gui = (!has_default && attrs.rgb_ae_attr == HL_DEFAULT) ? 0 : attrs.rgb_ae_attr;
```

Comment:
I think this can be simplified. once we get past the if statement above it shouldn't matter if it was set previously or not. so just `g->sg_gui =  attrs.rgb_ae_attr & ~ HL_DEFAULT;` (and similar below)